pkg/config/metadata/user_info_test.go

```diff
@@ -64,33 +80,28 @@ func TestUserInfoCall(t *testing.T) {
 func TestUserInfoCanceledContext(t *testing.T) {
 	ctrl := NewController(t)
 	defer ctrl.Finish()
+	ta := newTestArtifacts(ctrl)
 
-	authContextMock := NewMockAuthContext(ctrl)
-	idConfEval := NewMockIdentityConfigEvaluator(ctrl)
-	oidcEvaluator, _ := identity.NewOIDC(fmt.Sprintf("http://%s", authServerHost), &authCredMock{})
-	idConfEval.EXPECT().GetOIDC().Return(oidcEvaluator)
-	authContextMock.EXPECT().GetHttp().Return(nil)
-	authContextMock.EXPECT().GetResolvedIdentity().Return(idConfEval, nil)
+	ta.authCredMock.EXPECT().GetCredentialsFromReq(Any()).Return("", nil)
+	ta.idConfEvalMock.EXPECT().GetOIDC().Return(ta.newOIDC)
+	ta.authContextMock.EXPECT().GetHttp().Return(nil)
+	ta.authContextMock.EXPECT().GetResolvedIdentity().Return(ta.idConfEvalMock, nil)
+
+	ta.cancel()
+	_, err := ta.userInfo.Call(ta.authContextMock, ta.ctx)
 
-	ctx, cancel := context.WithCancel(context.TODO())
-	cancel()
-	userInfo := UserInfo{oidcEvaluator}
-	_, err := userInfo.Call(authContextMock, ctx)
 	assert.Error(t, err, "context canceled")
 }
 
 func TestUserInfoMissingOIDCConfig(t *testing.T) {
 	ctrl := NewController(t)
 	defer ctrl.Finish()
+	ta := newTestArtifacts(ctrl)
 
-	authContextMock := NewMockAuthContext(ctrl)
-	idConfEval := NewMockIdentityConfigEvaluator(ctrl)
-	oidcEvaluator, _ := identity.NewOIDC(fmt.Sprintf("http://%s", authServerHost), &authCredMock{})
-	otherOidcEvaluator, _ := identity.NewOIDC("http://wrongServer", &authCredMock{})
-	idConfEval.EXPECT().GetOIDC().Return(otherOidcEvaluator)
-	authContextMock.EXPECT().GetResolvedIdentity().Return(idConfEval, nil)
+	wrongOidc, _ := identity.NewOIDC(fmt.Sprintf("http://wrongServer"), ta.authCredMock)
+	ta.idConfEvalMock.EXPECT().GetOIDC().Return(wrongOidc)
```

Comment:
```suggestion
	otherOidcEvaluator, _ := identity.NewOIDC("http://other-oidc-server", ta.authCredMock)
	ta.idConfEvalMock.EXPECT().GetOIDC().Return(otherOidcEvaluator)
```

The subtle difference between "other" and "wrong" is that it wouldn't be possible to have this particular UserInfo config if there wasn't another valid OIDC config in identity. In other words, there must be 2 OIDC identity configs (both correct); it's the UserInfo metadata config that happens to be linked to one of those identity configs that is NOT the one that resolved the token.